src/proto/unit_test/parse_list_test.cpp

```diff
@@ -0,0 +1,52 @@
+#include <catch2/catch.hpp>
+
+#include <lbann/base.hpp>
+#include <lbann/proto/proto_common.hpp>
+
+#include <string>
+#include <vector>
+
+TEST_CASE("Testing parse_list", "[proto][utilities]")
+{
+  SECTION("execution_mode")
+  {
+    const std::vector<lbann::execution_mode> expected =
+      { lbann::execution_mode::training,
+        lbann::execution_mode::validation,
+        lbann::execution_mode::testing };
+
+    auto const answer =
+      lbann::parse_list<lbann::execution_mode>("train validate test");
+    CHECK(answer == expected);
+    CHECK(
+      lbann::parse_list<lbann::execution_mode>("")
+      == std::vector<lbann::execution_mode>{});
+    CHECK(
+      lbann::parse_list<lbann::execution_mode>(" ")
+      == std::vector<lbann::execution_mode>{});
+
+    CHECK_THROWS(
+      lbann::parse_list<lbann::execution_mode>("banana tuna salad"));
```

Comment:
Sounds gross.

Reply:
Yep, that's why I picked it. Should I
```cpp
#define CHECK_THROWS_UP(...) \
    CHECK_THROWS(__VA_ARGS__)
```
to help drive the point home? 😛